Lib/ufo2ft/__init__.py

```diff
@@ -415,6 +415,26 @@ def compileInterpolatableTTFsFromDS(
                 "designspace source '%s' is missing required 'font' attribute"
                 % getattr(source, "name", "<Unknown>")
             )
+        updateFamilyName = (source.familyName and
+                            source.font.info.familyName != source.familyName)
+        updateStyleName = (source.styleName and
+                           source.font.info.styleName != source.styleName)
+        if (updateStyleName or updateFamilyName):
+            source.font = source.font.__class__(source.font.path)
```

Comment:
you're assuming that the font has a path (was loaded from disk) which is not guaranteed, it may as well be built from code.
How about `copy.deepcopy`? I think ufoLib2 fonts support that, not sure defcon's

Reply:
also, you're copying the font object but you are still modifying the input designspace object in-place (setting a different `source.font`)..

Reply:
the method has an `inplace` parameter, you should try respect that when deciding whether to copy or not